File: js/functions/functions_arr.js
/**
 * 
 * @param {number[]} arr 
 * @returns {number}
 */
function sum_array(arr) {
    let sum = 0;
    for (const number of arr) {
        sum += number
    }
    return sum;
}

/**
 * 
 * @param {number[][]} arr 
 * @returns {number}
 */
function sum_nested_array(arr) {
    let sum = 0;
    for (const inner_array of arr) {
        for (const number of inner_array) {
            sum += number
        }
    }
    return sum;
}

File: python/functions/functions_arr.py

def sum_array(arr):
    sum_arr = 0
    for number in arr:
        sum_arr = sum_arr + number
    
    return sum_arr


def sum_nested_array(arr) :
    sum_arr = 0
    for inner_array in arr:
        for number in inner_array:
            sum_arr = sum_arr + number
    
    return sum_arr


def sum_nested_array_one_loop(arr) :
    sum_arr = 0

    index = 0

    while index < len(arr):
        if isinstance(arr[index], list): # check if arr[index] is a list of numbers
            for item in arr[index]:
                arr.append(item)
            # arr.append(*arr[index])
        else:
            sum_arr = sum_arr + arr[index]
        index = index + 1
    return sum_arr

sum_nested = sum_nested_array_one_loop([
    [1 , 2],
    [-1 , [5 , [7]]]
])

print(sum_nested)